File: backend/app/core/upgrades/curses/fragility_1.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class Fragility1(UpgradeImplementation):
    id = 'fragility_1'
    name = 'Fragility 1'
    description = 'Increased damage taken by 1.25%'
    rarity = 'rare'
    type = 'effect'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        appliers.effects.register_effect('protection', {'value': 0.0125})

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/curses/fragility_1.ts
import { Upgrade, type UpgradeDef, type DamageRef } from '../Upgrade'
import { RarityID, UpgradeTypeID } from '../../data/ID'

export const Fragility1Def: UpgradeDef = {
  id: "fragility_1",
  name: "Fragility 1",
  description: "Increased damage taken by 1.25%",
  rarity: RarityID.Rare,
  upgradeType: UpgradeTypeID.Effect,
  cost: 0,
  value: 0.0125,
  isMultiplier: true,
  effect: "protection",
  stackable: true,
  maxStacks: 3,
  curse: true,
}

export class Fragility1 extends Upgrade {
  onApply(): void {}

  modifyPlayerHurt(damage: DamageRef): void {
    damage.amount *= 1 + this.def.value!
  }
}
